### ISSUE TITLE
Nil stops updating highlighting when typing fast

### ISSUE THREAD

**Issue by xddxdd** [oxalica/nil#27]
I'm using VSCode + Nix IDE plugin + Nil. When I'm editing a nix file and typing fast, nil will stop updating the highlight results in the editor window.

When this happens, nil process still runs in the background, but CPU usage is zero.

This is a gdb stacktrace of nil when this happens:

<details>
<summary>Click to expand</summary>

```
Thread 19 (Thread 0x7f3ca1dee640 (LWP 2042578) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 18 (Thread 0x7f3ca1fef640 (LWP 2042577) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0c81656c in std::sys::unix::locks::futex_rwlock::RwLock::read_contended ()
#3  0x0000557e0c82f1d3 in nil::handler::diagnostics ()
#4  0x0000557e0c83fa97 in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#5  0x0000557e0c847efc in nil::server::Server::worker ()
#6  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#7  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#8  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#9  0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#10 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 17 (Thread 0x7f3ca21f0640 (LWP 2042576) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 16 (Thread 0x7f3ca23f1640 (LWP 2042575) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
--Type <RET> for more, q to quit, c to continue without paging--
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 15 (Thread 0x7f3ca25f2640 (LWP 2042574) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 14 (Thread 0x7f3ca27f3640 (LWP 2042573) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 13 (Thread 0x7f3ca29f4640 (LWP 2042572) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

--Type <RET> for more, q to quit, c to continue without paging--
Thread 12 (Thread 0x7f3ca2bf5640 (LWP 2042571) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 11 (Thread 0x7f3ca2df6640 (LWP 2042570) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 10 (Thread 0x7f3ca2ff7640 (LWP 2042569) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 9 (Thread 0x7f3ca31f8640 (LWP 2042568) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
--Type <RET> for more, q to quit, c to continue without paging--
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 8 (Thread 0x7f3ca33f9640 (LWP 2042567) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 7 (Thread 0x7f3ca35fa640 (LWP 2042566) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 6 (Thread 0x7f3ca37fb640 (LWP 2042565) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

--Type <RET> for more, q to quit, c to continue without paging--
Thread 5 (Thread 0x7f3ca39fc640 (LWP 2042564) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 4 (Thread 0x7f3ca3bfd640 (LWP 2042563) "Worker"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0c87a912 in crossbeam_channel::context::Context::with::{{closure}} ()
#4  0x0000557e0c87b937 in crossbeam_channel::flavors::list::Channel<T>::recv ()
#5  0x0000557e0c85ceb9 in crossbeam_channel::channel::Receiver<T>::recv ()
#6  0x0000557e0c847ee8 in nil::server::Server::worker ()
#7  0x0000557e0c86d628 in _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h5c52ba5448a93094E.llvm.11896235126976355698 ()
#8  0x0000557e0c86edef in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 3 (Thread 0x7f3ca3dfe640 (LWP 2042560) "nil"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0ca486f3 in crossbeam_channel::context::Context::wait_until ()
#4  0x0000557e0ca49010 in crossbeam_channel::context::Context::with::{{closure}} ()
#5  0x0000557e0ca4a576 in crossbeam_channel::flavors::zero::Channel<T>::send ()
#6  0x0000557e0ca46e74 in crossbeam_channel::channel::Sender<T>::send ()
#7  0x0000557e0ca4488e in std::sys_common::backtrace::__rust_begin_short_backtrace ()
#8  0x0000557e0ca45695 in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 2 (Thread 0x7f3ca3fff640 (LWP 2042559) "nil"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0ca769c5 in std::sys::unix::futex::futex_wait ()
#2  0x0000557e0ca78f59 in std::thread::park ()
#3  0x0000557e0ca486f3 in crossbeam_channel::context::Context::wait_until ()
#4  0x0000557e0ca488bd in crossbeam_channel::context::Context::with::{{closure}} ()
#5  0x0000557e0ca49eff in crossbeam_channel::flavors::zero::Channel<T>::recv ()
--Type <RET> for more, q to quit, c to continue without paging--
#6  0x0000557e0ca46f8c in crossbeam_channel::channel::Receiver<T>::recv ()
#7  0x0000557e0ca4449b in std::sys_common::backtrace::__rust_begin_short_backtrace ()
#8  0x0000557e0ca45825 in core::ops::function::FnOnce::call_once{{vtable.shim}} ()
#9  0x0000557e0ca80053 in std::sys::unix::thread::Thread::new::thread_start ()
#10 0x00007f3ca4088e86 in start_thread () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#11 0x00007f3ca410fc60 in clone3 () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6

Thread 1 (Thread 0x7f3ca42e6940 (LWP 2042558) "nil"):
#0  0x00007f3ca4107d3d in syscall () from /nix/store/ynn1by1qdl16q6qwwh2h7zkgrn36c6i8-glibc-2.35-163/lib/libc.so.6
#1  0x0000557e0c9abfdd in parking_lot::raw_rwlock::RawRwLock::wait_for_readers ()
#2  0x0000557e0c80d53e in parking_lot::raw_rwlock::RawRwLock::lock_exclusive_slow ()
#3  0x0000557e0c9a7552 in salsa::runtime::Runtime::with_incremented_revision ()
#4  0x0000557e0c9a7435 in salsa::runtime::Runtime::synthetic_write ()
#5  0x0000557e0c937a49 in ide::ide::AnalysisHost::apply_change ()
#6  0x0000557e0c84f949 in nil::server::Server::apply_vfs_change ()
#7  0x0000557e0c84af2e in nil::server::Server::run ()
#8  0x0000557e0c83456f in nil::main_loop ()
#9  0x0000557e0c81e9f0 in nil::main ()
#10 0x0000557e0c81fa53 in std::sys_common::backtrace::__rust_begin_short_backtrace ()
#11 0x0000557e0c823ca9 in _ZN3std2rt10lang_start28_$u7b$$u7b$closure$u7d$$u7d$17h55224661bd4805a4E.llvm.15921359885502866966 ()
#12 0x0000557e0ca78985 in std::rt::lang_start_internal ()
#13 0x0000557e0c81f038 in main ()
```

</details>

VSCode version: 1.72.2
Nix IDE plugin version: 0.2.1
Nil version: commit 93f15d8

**Comment by oxalica** [comment]
Should be fixed in a13e89c6229bcc2525ef3c8f3eeadbbce87f5374. Could you test it out?

**Comment by xddxdd** [comment]
Problem can no longer be reproduced. Thanks for your prompt fix!
